libs/core/langchain_core/tools/convert.py

```diff
@@ -421,3 +429,15 @@ def invoke_wrapper(callbacks: Optional[Callbacks] = None, **kwargs: Any) -> Any:
             description=description,
             args_schema=args_schema,
         )
+
+
+class MethodTool:
+    """A descriptor that converts a method into a tool."""
+
+    def __init__(self, func: Union[Callable, classmethod]) -> None:
+        self.func = func
+
+    def __get__(self, instance: Any, owner: Any) -> BaseTool:
+        if isinstance(self.func, classmethod):
+            return tool(self.func.__func__, outer_instance=owner)
+        return tool(self.func, outer_instance=instance)
```

Comment:
```suggestion
def toolmethod(f: callable) -> property:
    """A decorator that converts a method into a tool."""
    def inner_f(outer_instance) -> BaseTool:
        tool_f = functools.partial(f, outer_instance) # this substitutes in for self
        return tool(tool_f)
            
    return property(inner_f)
```

Reply:
this leaves out what was being handled by owner vs instance for `outer_instance` - what is that doing for us?

Reply:
That was to allow the support for classmethods. With the way the `__get__` descriptor works `owner` is for the class itself and `instance` is for the instance of the class.